docs/src/content/docs/cookbook/dexes/dedust.mdx

```diff
@@ -7,8 +7,434 @@ sidebar:
+    fulfillPayload: Cell? = null;
+
+    // Custom payload to attack to the transaction if the provisioning is rejected
+    // Defaults to `null`, which means no payload
+    rejectPayload: Cell? = null;
+}
+
+/// https://docs.dedust.io/reference/tlb-schemes#message-deposit_liquidity
+message(0xd55e4686) NativeDepositLiquidity {
+    // Unique identifier used to trace transactions across multiple contracts
+    // Defaults to 0, which means we don't mark messages to trace their chains
+    queryId: Int as uint64 = 0;
+
+    // Toncoin amount for the deposit
+    amount: Int as coins;
+
+    // Inlined fields of JettonDepositLiquidity Message without the opcode prefix
+    poolType: Int as uint1;
+    asset0: Asset;
+    asset1: Asset;
+    minimalLpAmount: Int as coins = 0;
+    targetBalances0: Int as coins;
+    targetBalances1: Int as coins;
+    fulfillPayload: Cell? = null;
+    rejectPayload: Cell? = null;
+}
+
+/// https://docs.dedust.io/reference/tlb-schemes#asset
+struct Asset {
+    // Specify 0 for native (TON) and omit all following fields
+    // Specify 1 for Jetton and then you must set non-null values for the following fields
+    type: Int as uint4;
+
+    workchain: Int? as uint8 = null;
+    address: Int? as uint256 = null;
+}
+
+const PoolTypeVolatile: Int = 0;
+const PoolTypeStable: Int = 1;
+
+const AssetTypeNative: Int = 0b0000;
+const AssetTypeJetton: Int = 0b0001;
+
+const JettonProvideLpGas: Int = ton("0.5");
+const JettonProvideLpGasFwd: Int = ton("0.4");
+const TonProvideLpGas: Int = ton("0.15");
+
+// This example directly uses the provided `myJettonWalletAddress`
+// In real-world scenarios, it's more reliable to calculate this address on-chain or save it during initialization to prevent any issues
+fun provideLiquidity(myJettonWalletAddress: Address) {
+    let jettonMasterRaw = parseStdAddress(
+        address("kQDkRHlWaibL7Tww48T6xAUFevPflca7i8TIiQTacBmnDOrb")
+        .asSlice()
+    );
+
+    // Step 1. Prepare input
+    let jettonAmount = ton("1");
+    let tonAmount = ton("1");
+
+    let asset0 = Asset{
+        type: AssetTypeNative,
+    };
+    let asset1 = Asset{
+        type: AssetTypeJetton,
+        workchain: jettonMasterRaw.workchain,
+        address: jettonMasterRaw.address,
+    };
+
+    // Step 2. Deposit Jetton to Vault
+    let jettonDepositBody = JettonDepositLiquidity{
+        poolType: PoolTypeVolatile,
+        asset0,
+        asset1,
+        targetBalances0: tonAmount,
+        targetBalances1: jettonAmount,
+    }.build(); // notice the .build() and not .toCell(),
+               // since we want some custom serialization logic!
+
+    send(SendParameters{
+        to: myJettonWalletAddress,
+        value: JettonProvideLpGas,
+        body: JettonTransfer{
+            queryId: 42,
+            amount: jettonAmount,
+            // Jetton Vault
+            destination: address("kQDcUuH4xhKejjilZAIeGuBh5JRWpzbVDcO9Qfh_Q_K4q9vk"),
```

Comment:
And this is real Jetton Vault for no-name jetton in testnet. Maybe consider using real usdt-related addresses? (Usdt Vault, Usdt JettonMaster, TON/USDT DeDust pool, etc.)

Reply:
Fixed all addresses